### PR TITLE
Update VC spec

### PR DESCRIPTION
**Pull request by przydatek**
Update VC specification, to account for the recent API adjustments and extensions.

### CODE REVIEW

File: docs/vc-spec.md [hunk 1]
```diff
@@ -11,60 +11,82 @@ The Candid interface is as follows, and the subsequent sections describe the
...
 type SignedIdAlias = record {
     credential_jws : text;
 };
-service : {
-    get_credential : (GetCredentialRequest) -> (GetCredentialResponse) query;
-    prepare_credential : (PrepareCredentialRequest) -> (PrepareCredentialResponse);
-    vc_consent_message : (Icrc21VcConsentMessageRequest) -> (Icrc21ConsentMessageResponse);
+type PreparedCredentialData = record { prepared_context : opt vec nat8 };
+
+/// Messages for `get_credential`.
```
Comment (frederikrothenberger) [inline docs/vc-spec.md:47]: I would use `Types` rather than `Messages` here?

File: docs/vc-spec.md [hunk 2]
```diff
@@ -124,27 +166,22 @@ consumes it.  For example, when using [canister signatures](https://internetcomp
...
 
 type GetCredentialRequest = record {
     signed_id_alias : SignedIdAlias;
-    prepared_context : opt blob;
     credential_spec : CredentialSpec;
-};
-type GetCredentialResponse = variant {
-    Ok : IssuedCredentialData;
-    Err : IssueCredentialError;
+    prepared_context : opt vec nat8;
```
Comment (frederikrothenberger) [inline docs/vc-spec.md:182]: Why did you change it to `vec nat8`? `blob` is an official candid shorthand for that.

File: docs/vc-spec.md [hunk 3]
```diff
@@ -11,60 +11,82 @@ The Candid interface is as follows, and the subsequent sections describe the
...
-    credential_spec : CredentialSpec;
-};
-type Icrc21ConsentMessageResponse = variant {
-    Ok : Icrc21ConsentInfo;
-    Err : Icrc21Error;
-};
 type Icrc21ConsentPreferences = record { language : text };
 type Icrc21Error = variant {
-    GenericError : Icrc21ErrorInfo;
+    GenericError : record { description : text; error_code : nat };
```
Comment (frederikrothenberger) [inline docs/vc-spec.md:27]: There is also `InsufficientPayment` now (if you are already updating these types).

Comment (przydatek) [reply docs/vc-spec.md:27]: I'll add it in a yet another PR, as this requires a change of code as well.

File: docs/vc-spec.md (hunk 1 above)
Comment (przydatek) [reply docs/vc-spec.md]: Done, here and in comments of the .did-file

File: docs/vc-spec.md (hunk 2 above)
Comment (przydatek) [reply docs/vc-spec.md]: Ah, copy-paste from .did-file, sorry!
Reverted the change.

**Review by przydatek — COMMENTED** [review]
Thanks, PTAL.

**Review by frederikrothenberger — APPROVED** [review]
LGTM, thanks!
